src/redis_stack/working_with_bloom_filter.md

```diff
@@ -0,0 +1,42 @@
+RedisBloom is developed by Redis Inc., and adds probabilistic data structures, including a Bloom filter, to Redis. To install RedisBloom on top of an existing Redis, download and run [Redis Stack](https://redis.io/docs/stack/get-started/install/), which includes RedisBloom and other capabilities. Also check out [RedisBloom commands](https://redis.io/commands/?group=bf).
+
+A Bloom filter is a probabilistic data structure that enables you to check if an element is present in a set using a very small memory space of a fixed size. It can guarantee the absence of an element from a set by giving a Boolean response about its presence. So, when it responds that an element is not present in a set ('false`), you can be sure that indeed is the case. However, false positive matches are possible. Controllable tradeoffs between accuracy and memory consumption from a Bloom filter are possible via the `error_rate` argument of the `BF.RESERVE` command.
+
+If you receive unhelpful queries from the clients to the database because no keys are matched in Redis, you can use a Bloom filter to filter such queries.
+
+These guidelines help you learn how to use a Bloom filter to reduce heavy calls to the relational database or memory.
+
+## Avoiding cache penetration
```

Comment:
Again, the assumption is that there is some heavy database (not Redis) in the system, and we want to use RedisBloom to avoid cache penetration for that external database (not for Redis itself).

It is important to explain this deployment here and the problem that we are trying to solve in order to make this use case understandable.

In this specific example each query is client's IP address and information about this IP address is stored in some external (presumably slow) database. There are many queries with IP addresses that do not exist in the external database, and we want to reduce the number of queries to the external database by detecting non-existing IP addresses using RedisBloom and hence reducing the number of accesses to the external DB.